swri_transform_util/nodes/initialize_origin.py

```diff
@@ -55,24 +55,40 @@ def gps_callback(data):
         
         _origin_pub.publish(_gps_fix)
 
+def gps_diagnostic_callback(data):
+    global _gps_fix
+    global _too_far_from_origin
+    
+    if _gps_fix != None:
+        # Compare the local_xy_origin gps fix to most recent gps value
+        # If difference is greater than 1 deg in lat/lon, we have traveled more than 10Km
+        # which will be reported as a warning since our near_field assumptions will fail.
+        if abs(data.latitude - _gps_fix.latitude) > 1.0 or abs(data.longitude - _gps_fix.longitude) > 1.0:
+          _too_far_from_origin = True
+        else:
+          _too_far_from_origin = False
+
 def initialize_origin():
     rospy.init_node('initialize_origin', anonymous=True)
    
     global _origin_pub
     global _local_xy_frame
+    global _too_far_from_origin
     _origin_pub = rospy.Publisher('/local_xy_origin', GPSFix, latch=True, queue_size=2)
     
     diagnostic_pub = rospy.Publisher('/diagnostics', DiagnosticArray, queue_size=2)
    
     local_xy_origin = rospy.get_param('~local_xy_origin', 'auto')
     _local_xy_frame = rospy.get_param('~local_xy_frame', 'map')
     _local_xy_frame_identity = rospy.get_param('~local_xy_frame_identity', _local_xy_frame + "__identity")
+    _too_far_from_origin = False
    
     if local_xy_origin == "auto":
         global _sub
         _sub = rospy.Subscriber("gps", GPSFix, gps_callback)
     else:
         parse_origin(local_xy_origin)
+    _gps_diag_sub = rospy.Subscriber("gps", GPSFix, gps_diagnostic_callback)
```

Comment:
What happens in rospy when you have two subscribers on the same topic? I've never tried it.